File: tests/test_edit_project.py
# -*- coding: utf-8 -*-
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.ui import Select
from selenium.common.exceptions import NoSuchElementException
from selenium.common.exceptions import NoAlertPresentException
import unittest, time, re

class TestSearchFilter(unittest.TestCase):
    def setUp(self):
        self.driver = webdriver.Firefox()
        self.driver.implicitly_wait(30)
        self.base_url = "http://caffeinestate.com/qa/client/"
        self.verificationErrors = []
        self.accept_next_alert = True
    
    def test_search_filter(self):
        driver = self.driver
        driver.get(self.base_url + "/qa/client/")
        driver.find_element_by_id("input_0").clear()
        driver.find_element_by_id("input_0").send_keys("delivered")

driver.find_element_by_xpath("(//img[@id='editIcon'])[2]").click()
driver.find_element_by_xpath("(//img[@id='editIcon'])[5]").click()
driver.find_element_by_xpath("(//img[@id='editIcon'])[5]").click()
driver.find_element_by_id("input_38").clear()
driver.find_element_by_id("input_38").send_keys("1200.00")
driver.find_element_by_xpath("(//img[@id='editIcon'])[6]").click()
driver.find_element_by_id("input_39").clear()
driver.find_element_by_id("input_39").send_keys("delivered - status")
driver.find_element_by_xpath("(//img[@id='editIcon'])[3]").click()
driver.find_element_by_xpath("(//button[@id='moreInfoButton'])[3]").click()
driver.find_element_by_id("input_0").clear()
driver.find_element_by_id("input_0").send_keys("")
    
    def is_element_present(self, how, what):
        try: self.driver.find_element(by=how, value=what)
        except NoSuchElementException as e: return False
        return True
    
    def is_alert_present(self):
        try: self.driver.switch_to_alert()
        except NoAlertPresentException as e: return False
        return True
    
    def close_alert_and_get_its_text(self):
        try:
            alert = self.driver.switch_to_alert()
            alert_text = alert.text
            if self.accept_next_alert:
                alert.accept()
            else:
                alert.dismiss()
            return alert_text
        finally: self.accept_next_alert = True
    
    def tearDown(self):
        self.driver.quit()
        self.assertEqual([], self.verificationErrors)

if __name__ == "__main__":
    unittest.main()

File: tests/test_edit_project.java
package com.example.tests;

import java.util.regex.Pattern;
import java.util.concurrent.TimeUnit;
import org.testng.annotations.*;
import static org.testng.Assert.*;
import org.openqa.selenium.*;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

public class TestSearchFilter {
  private WebDriver driver;
  private String baseUrl;
  private boolean acceptNextAlert = true;
  private StringBuffer verificationErrors = new StringBuffer();

  @BeforeClass(alwaysRun = true)
  public void setUp() throws Exception {
    driver = new FirefoxDriver();
    baseUrl = "http://caffeinestate.com/qa/client/";
    driver.manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);
  }

  @Test
  public void testSearchFilter() throws Exception {
    driver.get(baseUrl + "/qa/client/");
    driver.findElement(By.id("input_0")).clear();
    driver.findElement(By.id("input_0")).sendKeys("delivered");

driver.findElement(By.xpath("(//img[@id='editIcon'])[2]")).click();
driver.findElement(By.xpath("(//img[@id='editIcon'])[5]")).click();
driver.findElement(By.xpath("(//img[@id='editIcon'])[5]")).click();
driver.findElement(By.id("input_38")).clear();
driver.findElement(By.id("input_38")).sendKeys("1200.00");
driver.findElement(By.xpath("(//img[@id='editIcon'])[6]")).click();
driver.findElement(By.id("input_39")).clear();
driver.findElement(By.id("input_39")).sendKeys("delivered - status");
driver.findElement(By.xpath("(//img[@id='editIcon'])[3]")).click();
driver.findElement(By.xpath("(//button[@id='moreInfoButton'])[3]")).click();
driver.findElement(By.id("input_0")).clear();
driver.findElement(By.id("input_0")).sendKeys("");
  }

  @AfterClass(alwaysRun = true)
  public void tearDown() throws Exception {
    driver.quit();
    String verificationErrorString = verificationErrors.toString();
    if (!"".equals(verificationErrorString)) {
      fail(verificationErrorString);
    }
  }

  private boolean isElementPresent(By by) {
    try {
      driver.findElement(by);
      return true;
    } catch (NoSuchElementException e) {
      return false;
    }
  }

  private boolean isAlertPresent() {
    try {
      driver.switchTo().alert();
      return true;
    } catch (NoAlertPresentException e) {
      return false;
    }
  }

  private String closeAlertAndGetItsText() {
    try {
      Alert alert = driver.switchTo().alert();
      String alertText = alert.getText();
      if (acceptNextAlert) {
        alert.accept();
      } else {
        alert.dismiss();
      }
      return alertText;
    } finally {
      acceptNextAlert = true;
    }
  }
}
